File: app/src/main/java/com/example/weizhenbin/wangebug/base/AppActivityLifecycleListener.java
package com.example.weizhenbin.wangebug.base;

import android.app.Activity;
import android.os.Bundle;

/**
 * Created by weizhenbin on 2018/9/21.
 */

public abstract class AppActivityLifecycleListener {
    protected abstract void onActivityCreated(Activity activity, Bundle savedInstanceState);
    protected abstract void onActivityStarted(Activity activity);
    protected abstract  void onActivityResumed(Activity activity);
    protected abstract  void onActivityPaused(Activity activity);
    protected abstract  void onActivityStopped(Activity activity);
    protected abstract void onActivitySaveInstanceState(Activity activity, Bundle outState);
    protected abstract  void onActivityDestroyed(Activity activity);

    protected abstract void onAppForeground();
    protected abstract void onAppBackground();
}

File: app/src/main/java/com/example/weizhenbin/wangebug/base/AppActivityLifecycleListener.kt
package com.example.weizhenbin.wangebug.base

import android.app.Activity
import android.os.Bundle

/**
 * Created by weizhenbin on 2018/9/21.
 */

abstract class AppActivityLifecycleListener {
     abstract fun onActivityCreated(activity: Activity?, savedInstanceState: Bundle?)
     abstract fun onActivityStarted(activity: Activity?)
     abstract fun onActivityResumed(activity: Activity?)
     abstract fun onActivityPaused(activity: Activity?)
     abstract fun onActivityStopped(activity: Activity?)
     abstract fun onActivitySaveInstanceState(activity: Activity?, outState: Bundle?)
     abstract fun onActivityDestroyed(activity: Activity?)

     abstract fun onAppForeground()
     abstract fun onAppBackground()
}
